Expand allowable llvm versions from [7, 11.1] to [7, 12]

scober opened the pull request:
Also update failure message to accurately reflect the range of versions
searched for.

ddn0 commented on libava/manager/manager.cpp:
```diff
@@ -152,8 +152,16 @@ class DaemonServiceClient {
   }
 
+  bool IsDead() {
+    if (!is_dead_ && channel_->GetState(false) == GRPC_CHANNEL_SHUTDOWN)
```
qq: what was the behavior before adding this fix?

yuhc replied:
The manager might request an API server from a dead spawn daemon and the request would fail. Then the manager would return an "insufficient resource" error to the application.

It was assumed that the daemons always keep alive after they were registered in the manager. Any changes to the daemons' states would require to restart the manager and all the daemons. Now it's okay for the daemons to join and leave.

yuhc replied:
Previously a failed request still reserved the GPU memory, so that finally the dead daemon's GPU resources would be used up and no request would be sent to that daemon. It was a dirty implementation.

ddn0 replied:
... and technically applications should retry their request when they get an "insufficient resource" error, and before this change, if an application did retry, it would hit the same dead daemon and now it will eventually try a different, probably better daemon?

yuhc replied:
Mostly yes--the applications should retry/rerun when they get an "insufficient resource" error.
Before this change, the application would eventually hit a better daemon.
Now the application will get the error only in two rare cases:
1. The chosen spawn daemon fails to spawn the API server (it shouldn't be possible unless the daemon VM is corrupt, in which case the daemon VM should be re-installed).
2. The connection is valid when it's checked, but closed after that. The application should retry, but it won't hit the same daemon anymore.

Commit:
Expand allowable llvm versions from [7, 11.1] to [7, 12]

Also update failure message to accurately reflect the range of versions
searched for.

arthurp reviewed (approved):
LGTM.
